docarray/array/stacked/column_storage.py

```diff
@@ -0,0 +1,196 @@
+                        val = getattr(doc, field_name)
+                        if val is None:
+                            val = tensor_type.get_comp_backend().none_value()
+                        tf_stack.append(val.tensor)
+
+                    stacked: tf.Tensor = tf.stack(tf_stack)
+                    tensor_columns[field_name] = TensorFlowTensor(stacked)
+
+                elif issubclass(field_type, AbstractTensor):
+
+                    tensor = getattr(docs[0], field_name)
+                    column_shape = (
+                        (len(docs), *tensor.shape)
+                        if tensor is not None
+                        else (len(docs),)
+                    )
+                    tensor_columns[field_name] = field_type._docarray_from_native(
+                        field_type.get_comp_backend().empty(
+                            column_shape,
+                            dtype=tensor.dtype if hasattr(tensor, 'dtype') else None,
+                            device=tensor.device if hasattr(tensor, 'device') else None,
+                        )
+                    )
+
+                    for i, doc in enumerate(docs):
+                        val = getattr(doc, field_name)
+                        if val is None:
+                            val = tensor_type.get_comp_backend().none_value()
+
+                        cast(AbstractTensor, tensor_columns[field_name])[i] = val
+
+                elif issubclass(field_type, BaseDocument):
+                    doc_columns[field_name] = getattr(docs, field_name).stack()
+
+                elif issubclass(field_type, DocumentArray):
+                    da_columns[field_name] = list()
+                    for doc in docs:
+                        da_columns[field_name].append(getattr(doc, field_name).stack())
+                else:
+                    any_columns[field_name] = getattr(docs, field_name)
+            else:
+                any_columns[field_name] = getattr(docs, field_name)
+
+        return cls(
+            tensor_columns,
+            doc_columns,
+            da_columns,
+            any_columns,
+            document_type,
+            tensor_type,
+        )
+
+    def __len__(self) -> int:
+        return len(self.any_columns['id'])  # TODO what if ID are None ?
+
+    def __getitem__(self: T, item: IndexIterType) -> T:
+        tensor_columns = {key: col[item] for key, col in self.tensor_columns.items()}
+        doc_columns = {key: col[item] for key, col in self.doc_columns.items()}
+        da_columns = {key: col[item] for key, col in self.da_columns.items()}
+        any_columns = {key: col[item] for key, col in self.any_columns.items()}
+
+        return ColumnStorage(
+            tensor_columns,
+            doc_columns,
+            da_columns,
+            any_columns,
+            self.document_type,
+            self.tensor_type,
+        )
+
+
+class ColumnStorageView(dict, MutableMapping[str, Any]):
```

Comment:
is this double inheritance needed? isn't dict already a mutable mapping?

Reply:
Also, docstring plz :)

Reply:
docstring coming everywhere !

Reply:
You cannot do 
```python
class MyClass(dict[str, Any]):
   ...
```

and if you do only 
```
class MyClass(MutableMapping[str, Any]):
   ...
```

then you don't have dict features.


So if you want both type hint and dict behavior you need both I think ? Maybe using `Dict[str,Any]` could work.